File: RepoTransBench/target_projects/Java/C#/doyensec_ajpfuzzer/tests/original/UtilsTest.cs
using System;
using System.Collections.Generic;
using Xunit;
using Doyensec.Ajpfuzzer;

namespace Doyensec.Ajpfuzzer.Tests.Original
{
    public class UtilsTest
    {
        [Fact]
        public void TestRandomStringNegativeLength()
        {
            Assert.Throws<ArgumentException>(() => Utils.RandomString(-1, "abc"));
        }

        [Fact]
        public void TestRandomStringWithEmptyChars()
        {
            Assert.Throws<ArgumentException>(() => Utils.RandomString(5, ""));
        }

        [Fact]
        public void TestRandomStringWithValidInput()
        {
            string result = Utils.RandomString(8, "abcd");
            Assert.NotNull(result);
            Assert.Equal(8, result.Length);
            Assert.Matches("^[abcd]{8}$", result);
        }

        [Fact]
        public void TestGetRandomIntEdgeCases()
        {
            Assert.Equal(5, Utils.GetRandomInt(5, 5));
            Assert.Equal(42, Utils.GetRandomInt(42, 42));
        }

        [Fact]
        public void TestGetRandomIntRange()
        {
            for (int i = 0; i < 100; ++i)
            {
                int result = Utils.GetRandomInt(10, 20);
                Assert.InRange(result, 10, 20);
            }
        }

        [Fact]
        public void TestCreateMapFromPairsEvenArguments()
        {
            var map = Utils.CreateMapFromPairs<string, object>("k1", 123, "k2", "vv", "k3", null);
            Assert.Equal(3, map.Count);
            Assert.Equal(123, map["k1"]);
            Assert.Equal("vv", map["k2"]);
            Assert.Null(map["k3"]);
        }

        [Fact]
        public void TestCreateMapFromPairsOddArguments()
        {
            Assert.Throws<ArgumentException>(() => Utils.CreateMapFromPairs<string, object>("a", "b", "c"));
        }

        [Fact]
        public void TestToHex()
        {
            byte[] b = new byte[] { 0, 10, 15, 16, 31, 127, 255 };
            string hex = Utils.ToHex(b);
            Assert.Matches("^[0-9a-f]{14}$", hex);
            Assert.Equal("000a0f101f7fff", hex);
        }

        [Fact]
        public void TestToHexNull()
        {
            Assert.Null(Utils.ToHex(null));
        }

        [Fact]
        public void TestToHexEmpty()
        {
            Assert.Equal("", Utils.ToHex(new byte[0]));
        }

        [Fact]
        public void TestJoinSimple()
        {
            string[] arr = new string[] { "a", "b", "c" };
            string result = Utils.Join(arr, ":");
            Assert.Equal("a:b:c", result);
        }

        [Fact]
        public void TestJoinNull()
        {
            Assert.Equal("", Utils.Join(null, ","));
        }

        [Fact]
        public void TestJoinEmptyArray()
        {
            Assert.Equal("", Utils.Join(new string[0], "|"));
        }

        [Fact]
        public void TestJoinNoSeparator()
        {
            string[] arr = { "x", "y" };
            Assert.Equal("xy", Utils.Join(arr, ""));
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/doyensec_ajpfuzzer/tests/original/utils_test.go
package original

import (
	"regexp"
	"testing"

	"ajpfuzzer"
)

func TestRandomStringNegativeLength(t *testing.T) {
	defer func() {
		if r := recover(); r == nil {
			t.Error("Expected panic for negative length")
		}
	}()
	ajpfuzzer.RandomString(-1, "abc")
}

func TestRandomStringWithEmptyChars(t *testing.T) {
	defer func() {
		if r := recover(); r == nil {
			t.Error("Expected panic for empty chars")
		}
	}()
	ajpfuzzer.RandomString(5, "")
}

func TestRandomStringWithValidInput(t *testing.T) {
	res := ajpfuzzer.RandomString(8, "abcd")
	if len(res) != 8 {
		t.Errorf("Expected length 8, got %d", len(res))
	}
	match, _ := regexp.MatchString("^[abcd]{8}$", res)
	if !match {
		t.Errorf("Result %q does not match [abcd]{8}", res)
	}
}

func TestGetRandomIntEdgeCases(t *testing.T) {
	if ajpfuzzer.GetRandomInt(5, 5) != 5 {
		t.Errorf("Expected 5 for (5,5)")
	}
	if ajpfuzzer.GetRandomInt(42, 42) != 42 {
		t.Errorf("Expected 42 for (42,42)")
	}
}

func TestGetRandomIntRange(t *testing.T) {
	for i := 0; i < 100; i++ {
		res := ajpfuzzer.GetRandomInt(10, 20)
		if res < 10 || res > 20 {
			t.Errorf("Random value %d not in range 10..20", res)
		}
	}
}

func TestCreateMapFromPairsEvenArguments(t *testing.T) {
	m := ajpfuzzer.CreateMapFromPairs("k1", 123, "k2", "vv", "k3", nil)
	if m["k1"] != 123 {
		t.Errorf("Key k1 expected 123")
	}
	if m["k2"] != "vv" {
		t.Errorf("Key k2 expected 'vv'")
	}
	if m["k3"] != nil {
		t.Errorf("Key k3 expected nil, got %v", m["k3"])
	}
	if len(m) != 3 {
		t.Errorf("Expected map size 3")
	}
}

func TestCreateMapFromPairsOddArguments(t *testing.T) {
	defer func() {
		if r := recover(); r == nil {
			t.Error("Expected panic for odd number of args")
		}
	}()
	ajpfuzzer.CreateMapFromPairs("a", "b", "c")
}

func TestToHex(t *testing.T) {
	b := []byte{0, 10, 15, 16, 31, 127, 255}
	hex := ajpfuzzer.ToHex(b)
	match, _ := regexp.MatchString("^[0-9a-f]{14}$", hex)
	if !match {
		t.Errorf("Hex result %q does not match [0-9a-f]{14}", hex)
	}
	if hex != "000a0f101f7fff" {
		t.Errorf("Expected '000a0f101f7fff', got %s", hex)
	}
}

func TestToHexNull(t *testing.T) {
	if res := ajpfuzzer.ToHex(nil); res != "" {
		t.Errorf("Expected empty string for nil, got %q", res)
	}
}

func TestToHexEmpty(t *testing.T) {
	if res := ajpfuzzer.ToHex([]byte{}); res != "" {
		t.Errorf("Expected empty string for empty byte array, got %q", res)
	}
}

func TestJoinSimple(t *testing.T) {
	arr := []string{"a", "b", "c"}
	result := ajpfuzzer.Join(arr, ":")
	if result != "a:b:c" {
		t.Errorf("Expected 'a:b:c', got %q", result)
	}
}

func TestJoinNull(t *testing.T) {
	if ajpfuzzer.Join(nil, ",") != "" {
		t.Errorf("Expected empty string for nil input")
	}
}

func TestJoinEmptyArray(t *testing.T) {
	if ajpfuzzer.Join([]string{}, "|") != "" {
		t.Errorf("Expected empty string for empty array")
	}
}

func TestJoinNoSeparator(t *testing.T) {
	arr := []string{"x", "y"}
	if ajpfuzzer.Join(arr, "") != "xy" {
		t.Errorf("Expected 'xy', got %q", ajpfuzzer.Join(arr, ""))
	}
}